torch_frame/data/dataset.py

```diff
@@ -335,15 +342,17 @@ def materialize(self, device: Optional[torch.device] = None,
 
         # 1. Fill column statistics:
         for col, stype in self.col_to_stype.items():
-            if stype == torch_frame.multi_categorical:
-                ser = self.df[col].apply(
-                    lambda x: [cat.strip() for cat in x.split(self.sep)]
-                    if x is not None and x != '' else [])
-                ser = ser.explode()
-                ser = ser.dropna()
-            else:
-                ser = self.df[col]
-            self._col_stats[col] = compute_col_stats(ser, stype)
+            ser = self.df[col]
+            if stype == torch_frame.multicategorical and (self.sep is None or
+                                                          col not in self.sep):
+                warnings.warn(
+                    "The separator is not specified for multicategorical"
+                    " column: {col}. ',' will be set as default separator.")
+                if self.sep is None:
+                    self.sep = {}
+                self.sep[col] = ','
+            self._col_stats[col] = compute_col_stats(
+                ser, stype, sep=self.sep.get(col, None))
```

Comment:
Remove this and put it in init

Reply:
Also, do we need warning? Just logging seems enough to me.

Reply:
Also, do we need warning if in the documentation, we clearly state that the default is ","?

Reply:
I think it's good to have a warning for user.